src/VisualStudio/Core/Def/Implementation/Diagnostics/DiagnosticProgressReporter.cs

```diff
@@ -0,0 +1,139 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.ComponentModel.Composition;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.SolutionCrawler;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.TaskStatusCenter;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.Diagnostics
+{
+    [Export(typeof(DiagnosticProgressReporter))]
+    internal class DiagnosticProgressReporter
+    {
+        private readonly IVsTaskStatusCenterService _taskCenterService;
+        private readonly IDiagnosticService _diagnosticService;
+        private readonly TaskHandlerOptions _options;
+
+        // this fields are never accessed concurrently
+        private TaskCompletionSource<bool> _currentTask;
```

Comment:
:bulb: Use `TaskCompletionSource<VoidResult>` so it's clear that the value of the result has no meaning.

Reply:
where the VoidResult is?

Reply:
You can defined it as a struct in this type:

```csharp
private struct VoidResult
{
}
```